fsspec/spec.py

```diff
@@ -115,6 +118,59 @@ class AbstractFileSystem(metaclass=_Cached):
     #: Extra *class attributes* that should be considered when hashing.
     _extra_tokenize_attributes = ()
 
+    # Set by _Cached metaclass
+    storage_args: Tuple[Any, ...]
+    storage_options: Dict[str, Any]
+
+    class _JSONEncoder(json.JSONEncoder):
+        def default(self, o: Any) -> Any:
+            if isinstance(o, AbstractFileSystem):
+                return o.to_dict()
+            if isinstance(o, PurePath):
+                cls = type(o)
+                return {"cls": f"{cls.__module__}.{cls.__name__}", "str": str(o)}
```

Comment:
This is effectively pickling and a potential security hole (deser requires only an import, not an execute)

Reply:
Could you elaborate? Unlike pickling, this does not allow arbitrary code execution since the object needs to be importable in the first place.

Reply:
Importing a module executes code. Now, most python packages don't have side-effects on import, and of course only installed packages are importable. Furthermore, the code very reasonably checks for if `issubclass(path_cls, PurePath)` when loading. I would suggest a single sentence in the load-json docstring saying that it will do imports.

We do this kind of thing all the time (e.g., when creating filesystem instances by looking in the registry), but it's nice to be more explicit about this kind of thing.

Reply:
That sounds good, thanks for pointing that out!